packages/app/pages/api/v1/bhashachatbot/webhook.ts

```diff
@@ -0,0 +1,265 @@
+
+    console.log("Extracted important variables", waID, message, textBody, messageId);
+    if(waID === undefined || message === undefined){
+        res.status(403).json({
+            message: "It is a Status"
+        });
+        return ;
+    }
+
+    let parsedMessageId: MessageIdObj = {};
+
+    try {
+        parsedMessageId = JSON.parse(messageId);
+    }
+    catch (e) {
+        console.log("could not parse the id string", "treating as normal message");
+    }
+
+    console.log("Message Type", parsedMessageId.type);
+
+    const assigneDetails = await db.member.findFirst({
+        where: {
+            phone: `+${waID}`,
+        }
+    });
+
+    console.log("Fetched assignee details");
+
+    if (assigneDetails === null) {
+        console.log("user not registered");
+
+        await sendTextMessage(waID, "You are not registered. Please register your whats app number")
+
+        res.status(200).json({
+            message: "User not registered"
+        });
+
+        return;
+    }
+
+    let user_session = await db.user_session.findFirst({
+        where: {
+            member_id: assigneDetails.id
+        }
+    })
+    console.log("Searched for user session. Found : ", !!user_session);
+
+    if (!user_session) {
+        console.log("user session not found");
+        user_session = await db.user_session.create({
+            data: {
+                member_id: assigneDetails.id,
+            }
+        })
+    }
+
+    console.log("User Session", user_session)
+
+    if (textBody === "Hi") {
+        try {
+            if(!user_session.has_accepted_policy) {
+                await sendInteractiveMessage(waID, {
+                    body: {
+                        text: "Welcome to Bhasha ChatBot! Please click on below link to accept security policy https://www.whatsapp.com/legal/privacy-policy-eea"
+                    },
+                    type: "button",
+                    action: {
+                        buttons: [
+                            {
+                                type: "reply",
+                                reply: {
+                                    title: "I agree",
+                                    id: JSON.stringify({ type: "PA" }),
+                                }
+                            }
+                        ]
+                    }
+                });
+                res.status(200).json({
+                    message: "Policy sent for acceptance"
+                });
+                return;
+            }
+
+            await handleHiResponse(waID, assigneDetails, user_session)
+            res.status(200).json({
+                message: "Response of Hi - successfull."
+            });
+
+            return;
+        } catch (e) {
+            console.log("Error while handling Hi response :", e);
+            return;
+        }
+    }
+
+    if (message?.type === "text" && textBody !== "Hi" && user_session.current_question_uuid) {
```

Comment:
also fix `textBody !== "Hi"` with case insensitive version here, you can use toLowerCase and then compare to 'hi'

Reply:
Done